silkworm/db/datastore/snapshots/elias_fano/elias_fano_decoder.hpp

```diff
@@ -26,9 +26,15 @@ struct EliasFanoDecoder : public snapshots::Decoder {
 
     ~EliasFanoDecoder() override = default;
 
-    void decode_word(Bytes& word) override {
-        value = EliasFanoList32::from_encoded_data(word);
-        value.data_holder = std::move(word);
+    void decode_word(Word& word) override {
+        if (word.holds_bytes()) {
+            auto word_value = std::get<Bytes>(word);
+            value = EliasFanoList32::from_encoded_data(word_value);
+            value.data_holder = std::move(word_value);
```

Comment:
it would be nice to encapsulate this in EliasFanoList32. how about adding `from_encoded_data(Bytes)` that does exactly these 2 lines?